saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/OpenSamlAuthenticationRequestFactory.java

```diff
@@ -160,17 +166,30 @@ public Saml2RedirectAuthenticationRequest createRedirectAuthenticationRequest(
 
 	private AuthnRequest createAuthnRequest(Saml2AuthenticationRequestContext context) {
 		return createAuthnRequest(context.getIssuer(), context.getDestination(),
-				context.getAssertionConsumerServiceUrl(), this.protocolBindingResolver.convert(context));
+				context.getAssertionConsumerServiceUrl(), this.protocolBindingResolver.convert(context),
+				context.getRequestAuthnContextRefs(), context.getIsPassive(), context.getForceAuthn());
 	}
 
 	private AuthnRequest createAuthnRequest(String issuer, String destination, String assertionConsumerServiceUrl,
-			String protocolBinding) {
+			String protocolBinding, List<String> requestAuthnContextRefs, boolean isPassive, boolean forceAuth) {
 		AuthnRequest auth = this.authnRequestBuilder.buildObject();
 		auth.setID("ARQ" + UUID.randomUUID().toString().substring(1));
 		auth.setIssueInstant(new DateTime(this.clock.millis()));
-		auth.setForceAuthn(Boolean.FALSE);
-		auth.setIsPassive(Boolean.FALSE);
 		auth.setProtocolBinding(protocolBinding);
+		auth.setForceAuthn(forceAuth);
+		auth.setIsPassive(isPassive);
+		if (requestAuthnContextRefs != null) {
+			RequestedAuthnContext requestAuthnContext = new RequestedAuthnContextBuilder().buildObject();
```

Comment:
Let's read the builders from `XMLObjectProviderFactorySupport`. That way, if the application has a custom builder, then this class will use it. You can see how this is done with the other builders in the constructor.